src/io/FileSystem.js

```diff
@@ -0,0 +1,51 @@
+import fs from 'fs';
+import { PRODUCTS_ROUTE, PROMOTIONS_ROUTE } from '../constants/route.js';
+
+class FileSystem {
+  static getStocks() {
+    const file = fs.readFileSync(PRODUCTS_ROUTE, 'utf8');
+    return FileSystem.#processFile(file);
+  }
+
+  static getPromotions() {
+    const file = fs.readFileSync(PROMOTIONS_ROUTE, 'utf8');
+    return FileSystem.#processFile(file);
+  }
+
+  static writeProducts(rawStocks) {
+    const productsFile = this.#convertToCSV(rawStocks);
+    fs.writeFileSync(PRODUCTS_ROUTE, productsFile);
+  }
+
+  static #processFile(file) {
+    const splittedFile = file.split('\n');
+    const header = splittedFile[0].split(',');
+    const body = splittedFile.slice(1, -1);
+
+    const template = header.reduce((acc, cur) => {
+      acc[cur] = null;
+      return acc;
+    }, {});
+
+    return body.map((stockToken) =>
+      Object.keys(template).reduce((obj, key, index) => {
+        obj[key] = stockToken.split(',')[index];
+        return obj;
+      }, {}),
+    );
```

Comment:
Object.keys(template)이 header 배열과 같은 거라고 봐서 template을 만든 게 굳이 싶었어요.
중간에 template을 만든 이유가 궁금해요.

Reply:
reduce 내에서 stockToken.split(',')을 여러 번 하게 될 수 있으므로 한 번 split하고 재활용하는 건 어떤지 싶습니다:)

```suggestion
    return body.map((stockToken) => {
      const splittedStockToken = stockToken.split(‘,’);
      
      return Object.keys(template).reduce((obj, key, index) => {
        obj[key] = splittedStockToken[index];
        return obj;
      }, {}),
    });
```